server/routes/api/v1/users.js

```diff
@@ -220,18 +221,29 @@ router.post('/:id/invitations',
   (req, res) => {
     if (!req.body.project) return res.status(400).json(errorToJson(new Error('You need to specify the project you want to invite this user to')));
 
-    // TODO ensure that project and invited user exist
-    let joinRequest = new JoinRequests({
-      owner: req.user,
-      invited: req.params.id,
-      project: req.body.project
-    });
-    if (req.body.message) joinRequest.message = req.body.message;
-
-    joinRequest.save((err) => {
-      if (err) return res.status(500).json(errorToJson(err));
-      res.json(joinRequest);
-    });
+    Promise.all([
+      Projects.findById(req.body.project).exec(),
+      Users.findById(req.params.id).exec()
+    ])
+      .then(data => {
+        if (!data[0] || !data[1]) throw new Error('Your request was not ok. Either the project or the user don\'t exist.');
+
+        let joinRequest = new JoinRequests({
+          owner: req.user,
+          project: data[0],
+          invited: data[1]
+        });
+        if (req.body.message) joinRequest.message = req.body.message;
+
+        return Promise.all([
+          joinRequest.save(),
+          Notifications.create({message: `You have been invited to project ${data[0].name}`, owner: req.params.id})
```

Comment:
For easy navigation we can add this `a` tag to link to the project
```suggestion
          Notifications.create({message: `You have been invited to project <a href="/project/${data[0]._id}">${data[0].name}</a>`, owner: req.params.id})
```